src/ScrollView/ScrollView.tsx

```diff
@@ -0,0 +1,159 @@
+import * as React from 'react';
+import * as ReactDOM from 'react-dom';
+import * as cx from 'classnames';
+import { props, t, ReactChildren } from '../utils';
+import  GeminiScrollbar = require('gemini-scrollbar');
+import ResizeSensor from '../ResizeSensor/ResizeSensor';
+
+export type ScrollViewDefaultProps<CP extends React.HTMLAttributes<any>, ICP extends React.HTMLAttributes<any>> = {
+  /** component to use as the wrapper */
+  component: keyof React.ReactHTML | React.ComponentType<CP>,
+  /** props to pass to the wrapper component */
+  componentProps: CP,
+  /** component to use as the inner wrapper */
+  innerComponent: keyof React.ReactHTML | React.ComponentType<ICP>,
+  /** props to pass to the inner wrapper component */
+  innerComponentProps: ICP,
+  /** force ScrollView to use `gemini-scrollbar`s */
+  forceGemini: boolean,
+  /** style to pass to the wrapper component */
+  style: React.CSSProperties
+};
+
+export type ScrollViewRequiredProps = {
+  /** what to render inside the scroll view */
+  children: any,
+  /** whether to automatically show scrollbars */
+  autoshow?: boolean,
+  /** className to pass to the wrapper component */
+  className?: string,
+}
+
+export type ScrollViewState = {
+  isVerticalScrollbarVisible: boolean,
+  isHorizontalScrollbarVisible: boolean
+};
+
+export type ScrollViewProps<CP, ICP> = ScrollViewRequiredProps & Partial<ScrollViewDefaultProps<CP, ICP>>;
+export type ScrollViewDefaultedProps<CP, ICP> = ScrollViewRequiredProps & ScrollViewDefaultProps<CP, ICP>;
+
+export const Props = {
+  children: ReactChildren,
+  autoshow: t.maybe(t.Boolean),
+  forceGemini: t.maybe(t.Boolean),
+  component: t.maybe(t.union([t.Function, t.String])),
+  componentProps: t.maybe(t.Object),
+  innerComponent: t.maybe(t.union([t.Function, t.String])),
+  innerComponentProps: t.maybe(t.Object),
+  className: t.maybe(t.String),
+  style: t.maybe(t.Object)
+};
+
+/** A scrollable view */
+@props(Props)
+export default class ScrollView<CP extends React.HTMLAttributes<any>, ICP extends React.HTMLAttributes<any>> extends React.PureComponent<ScrollViewProps<CP, ICP>, ScrollViewState> {
+
+  static defaultProps: ScrollViewDefaultProps<React.HTMLAttributes<HTMLDivElement>, React.HTMLAttributes<HTMLDivElement>> = {
+    component: 'div',
+    componentProps: {},
+    forceGemini: true,
+    innerComponent: 'div',
+    innerComponentProps: {},
+    style: {}
+  }
+
+  state = {
+    isVerticalScrollbarVisible: false,
+    isHorizontalScrollbarVisible: false
+  }
+
+  /**
+   * Holds the reference to the GeminiScrollbar instance.
+   * @property scrollbar <public> [Object]
```

Comment:
this documentation comment seems useless and with an old format. I would remove it entirely.